app/views/posts/partials/show/_image.html.erb

```diff
@@ -1,3 +1,7 @@
 <% if policy(post).visible? %>
-  <%= image_tag(post.file_url_for(CurrentUser.user), width: post.image_width_for(CurrentUser.user), height: post.image_height_for(CurrentUser.user), id: "image", class: "fit-width", "data-original-width": post.image_width, "data-original-height": post.image_height, "data-large-width": post.large_image_width, "data-large-height": post.large_image_height, "data-tags": post.tag_string, alt: post.presenter.humanized_essential_tag_string, "data-uploader": post.uploader.name, "data-rating": post.rating, "data-flags": post.status_flags, "data-parent-id": post.parent_id, "data-has-children": post.has_children?, "data-has-active-children": post.has_active_children?, "data-score": post.score, "data-fav-count": post.fav_count, "itemprop": "contentUrl") %>
+  <%= content_tag(:picture) do -%>
+    <%= tag.source media: "(min-width: 660px)", srcset: post.tagged_file_url -%>
+    <%= tag.source media: "(max-width: 660px)", srcset: post.tagged_large_file_url -%>
```

Comment:
Isn't this a problem if [`post.tagged_large_file_url` doesn't actually exist](https://github.com/danbooru/danbooru/blob/82c09858f2cafa0ea1636251ad0c9341ff4dcfc7/app/models/post.rb#L230-L234)? I guess the browser would get a 404 and fall back on the `<img>` below but still, why even let it try?

Reply:
The tagged URLs will return the non-tagged URLs if that option is not enabled.

Reply:
Yeah, sorry. I didn't actually check and assumend [this check](https://github.com/danbooru/danbooru/blob/6d78d22571fbaa637e1a97a1aa1a9f39a34e0e09/app/models/post.rb#L173) was there for a reason and that `StorageManager#file_url` would behave [like `#file_path`](https://github.com/danbooru/danbooru/blob/6d78d22571fbaa637e1a97a1aa1a9f39a34e0e09/app/logical/storage_manager.rb#L85), [which it doesn't](https://github.com/danbooru/danbooru/blob/6d78d22571fbaa637e1a97a1aa1a9f39a34e0e09/app/logical/storage_manager.rb#L62)